test/bc_concretizations.jl

```diff
@@ -0,0 +1,158 @@
+using SparseArrays, DiffEqOperators, LinearAlgebra, Random,
+    Test, BandedMatrices, FillArrays, LazyArrays, BlockBandedMatrices, Compat
+
+@testset "Concretizations of BCs" begin
+    T = Float64
+    L = 10one(T)
+    N = 9
+    δx = L/(N+1)
+
+    @testset "Affine BCs" begin
+        @testset "Dirichlet0BC" begin
+            Q = Dirichlet0BC(T)
+
+            correct = vcat(zeros(T,1,N),
+                           Diagonal(ones(T,N)),
+                           zeros(T,1,N))
+
+            @testset "$mode concretization" for (mode,Mat,Expected,ExpectedBandwidths) in [
+                ("sparse -> Banded", sparse, BandedMatrix{T}, (1,-1)),
+                ("Banded", BandedMatrix, BandedMatrix{T}, (1,-1)),
+                ("Sparse", SparseMatrixCSC, SparseMatrixCSC{T}, nothing),
+                ("Dense", Array, Matrix{T}, nothing)
+            ]
+                Qm,Qu = Mat(Q,N)
+
+                @test Qm == correct
```

Comment:
Maybe add tests for the types you get out?

Reply:
That's exactly what I do on the next line, `@test Qm isa Expected`